c7n/resources/sns.py

```diff
@@ -44,4 +48,75 @@ class resource_type(object):
 
 @SNS.filter_registry.register('cross-account')
 class SNSCrossAccount(CrossAccountAccessFilter):
+    """Filter to return all SNS topics with cross account access permissions
+
+    The whitelist parameter will omit the accounts that match from the return
+
+    :example:
+
+        .. code-block:
+
+            policies:
+              - name: sns-cross-account
+                resource: sns
+                filters:
+                  - type: cross-account
+                    whitelist:
+                      - permitted-account-01
+                      - permitted-account-02
+    """
     permissions = ('sns:GetTopicAttributes',)
+
+
+@SNS.action_registry.register('remove-statements')
+class RemovePolicyStatement(RemovePolicyBase):
+    """Action to remove policy statements from SNS
+
+    :example:
+
+        .. code-block: yaml
+
+           policies:
+              - name: sns-cross-account
+                resource: sns
+                filters:
+                  - type: cross-account
+                actions:
+                  - type: remove-statements
+                    statement_ids: matched
+    """
+
+    permissions = ('sns:SetTopicAttributes', 'sns:GetTopicAttributes')
+
+    def process(self, resources):
+        results = []
+        client = local_session(self.manager.session_factory).client('sns')
+        for r in resources:
+            try:
+                if self.process_resource(client, r):
+                    results.append(r)
+            except:
+                self.log.exception(
+                    "Error processing sns:%s", r['TopicArn'])
+        return results
+
+    def process_resource(self, client, resource):
+        p = resource.get('Policy')
+        if p is None:
```

Comment:
this is  suspect, if the policy is not found, then it needs to be fetched or the action needs to require a policy filter as part of validation.

Reply:
Ah, I thought that since the enum spec of SNS is `get_topic_attributes` that if Policy wasn't there then it would be a duplicate call that won't find anything.

Reply:
good point, forgot that was in detail_spec for topics. feels a little debatable if should be tbh, lots of sns topics, api call per each is fairly time consuming. but thats not for this pr.